src/pypromice/qc/compute_percentiles.py

```diff
@@ -0,0 +1,370 @@
+                            exe_list
+                            )
+                    elif v=='t_u':
+                        df.set_index(timestamp, inplace=True)
+                        # data.drop(['time'], axis=1, inplace=True) # optionally drop original time column
+
+                        winter = df.t_u[df.index.month.isin([12,1,2])]
+                        spring = df.t_u[df.index.month.isin([3,4,5])]
+                        summer = df.t_u[df.index.month.isin([6,7,8])]
+                        fall = df.t_u[df.index.month.isin([9,10,11])]
+
+                        # Equivalent to above
+                        # winter = df.t_u[df.index.month.isin([12,1,2])]
+                        # spring = df.t_u[(df.index.month >= 3) & (df.index.month <= 5)]
+                        # summer = df.t_u[(df.index.month >= 6) & (df.index.month <= 8)]
+                        # fall = df.t_u[(df.index.month >= 9) & (df.index.month <= 11)]
+
+                        season_list = [winter,spring,summer,fall]
+                        season_integers = [1,2,3,4]
+
+                        for s, s_i in zip(season_list,season_integers):
+                            exe_list = [stid,s_i] # initialize list
+                            for i in quantiles:
+                                exe_list.append(s.quantile(q=i))
+                            exe_list.append(years)
+                            cur.execute(
+                                f'insert into {v} '
+                                '(stid,season,p0,p0p5,p1,p5,p10,p90,p95,p99,p99p5,p100,years) '
+                                'values (?,?,?,?,?,?,?,?,?,?,?,?,?)',
+                                exe_list
+                                )
+
+def _analyze_percentiles():
+    '''
+    This is run ONLY to examine percentile thresholds with context to full datasets
+    '''
+    for x in os.walk(args.l3_filepath):
+        if (len(x[2]) > 0): # files are present
+            stid = x[0].split(os.sep)[-1]
+           
+            
+            csv_file = [s for s in x[2] if '_hour.csv' in s]
+            if (len(csv_file) > 0) and (stid not in disclude_stations): # csv file is present
+                print(stid)
+                csv_filepath = x[0] + '/' + csv_file[0]
+                df = pd.read_csv(csv_filepath)
+                timestamp = pd.to_datetime(df.time)
+                df.set_index(timestamp, inplace=True)
+                _percentileQC(df, stid)
+
+def _percentileQC(df, stid):
+    '''
+    This is the same function that is found in L1toL2.py
+    Once thresholds are determined, they can be transferred to L1toL2.py
+    '''
+    # Define threshold dict to hold limit values, and 'hi' and 'lo' percentile.
+    # Limit values indicate how far we will go beyond the hi and lo percentiles to flag outliers.
+    # *_u are used to calculate and define all limits, which are then applied to *_u, *_l and *_i
+    var_threshold = {
```

Comment:
I am noticing that the values in this `var_threshold` are different that the values in `var_threshold` in `percentileQC`. Is this intentional? Was this function used for testing and determining thresholds?

Reply:
Thanks for spotting that, I will change it back! I tested some different thresholds and apparently did not change them back.